src/Nethermind/Nethermind.Evm/TransactionSubstate.cs

```diff
@@ -47,24 +47,32 @@ public TransactionSubstate(EvmExceptionType exceptionType, bool isTracerConnecte
                 {
                     if (Output.Length > 0)
                     {
-                        try
-                        {
-                            BigInteger start = Output.Span.Slice(4, 32).ToUnsignedBigInteger();
-                            BigInteger length = Output.Slice((int)start + 4, 32).Span.ToUnsignedBigInteger();
-                            Error = string.Concat("Reverted ",
-                                Output.Slice((int)start + 32 + 4, (int)length).ToArray().ToHexString(true));
-                        }
-                        catch (Exception)
+                        var span = Output.Span;
+                        if (span.Length > 32 + 4)
                         {
                             try
                             {
-                                Error = string.Concat("Reverted ", Output.ToArray().ToHexString(true));
+                                BigInteger start = span.Slice(4, 32).ToUnsignedBigInteger();
+                                BigInteger length = span.Slice((int)start + 4, 32).ToUnsignedBigInteger();
+                                Error = string.Concat("Reverted ",
+                                    Output.Slice((int)start + 32 + 4, (int)length).ToArray().ToHexString(true));
```

Comment:
Can we just check if:
a) Length is smaller than `Array.MaxLength`, or maybe other sane value (how many MB's should we allow to allocate here?)
b) Output has enough items to take

We could also skip `ToArray` and go directly to string from span or memory?